### PR TITLE
Update the ksops docs

### PR DESCRIPTION
**Pull request by hemajv**
Updating the ksops documentation

cc @HumairAK @anishasthana

### CODE REVIEW

**Comment by HumairAK** [comment]
Hey @hemajv can we change "Updating Private key" to `Updating Private key (For SRE Admins Only)` to separate the rest of the docs from regular users. 

Other than that, lgtm!

File: docs/secret_management.md [hunk 1]
```diff
@@ -1,11 +1,128 @@
...
+- [kustomize](https://github.com/kubernetes-sigs/kustomize/) built with Go (See [details below](#kustomize-go-plugin-caveats))
+- [SOPS](https://github.com/mozilla/sops)
+- gpg
+
+### 0. Verify Requirements
+Before continuing, verify your installation of [Go](https://github.com/golang/go), [SOPS](https://github.com/mozilla/sops), and `gpg`. Below are a few non-comprehensive commands to quickly check your installations:
+
+```bash
+# Verify Go is installed and your $GOPATH is set
+go env
```
Comment (anishasthana) [inline docs/secret_management.md:20]: I initially ran into an issue due to having a version of go that was too old -- we may want to make a callout here for that.

Comment (hemajv) [reply docs/secret_management.md:20]: updated

**Comment by HumairAK** [comment]
:ship:

**Pull request merged**